finagle-http/src/main/scala/com/twitter/finagle/http/codec/ConnectionManager.scala

```diff
@@ -11,7 +11,7 @@ import com.twitter.util.{Future, Promise}
 private[finagle] class ConnectionManager {
   private[this] var isKeepAlive = false
   private[this] var isIdle = true
-  private[this] var activeStreams = 0
+  private[this] var activeStreams, requestsObserved, responsesObserved = 0
```

Comment:
alternatively, we could track this as a single `pendingResponses` field (much like `activeStreams`).  I suppose this would be slightly better as it has no hope of overflowing